custom/conf/app.example.ini

```diff
@@ -379,6 +379,35 @@ INTERNAL_TOKEN=
 ;; Validate against https://haveibeenpwned.com/Passwords to see if a password has been exposed
 ;PASSWORD_CHECK_PWN = false
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+[security.password_hash]
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Every parameter for the built in hash algorithms can be changed in this section.
+;; Handle with care! Changing the values can massively change the hash calculation time and/or memory needed for the hashing process.
+;; After changing a value, the user password hash will be recalculated on next successful login.
+;;
+;; Parameters for BCrypt
+BCRYPT_COST = 10
+;;
+;; Parameters for scrypt
+SCRYPT_N = 65536
+SCRYPT_R = 16
+SCRYPT_P = 2
+SCRYPT_KEY_LENGTH = 50
+;;
+;; Parameters for Argon2id
+ARGON2_ITERATIONS = 2
+ARGON2_MEMORY = 65536
+ARGON2_PARALLELISM = 8
+ARGON2_KEY_LENGTH = 50
+;;
+;; Parameters for Pbkdf2
+PBKDF2_ITERATIONS = 10000
+PBKDF2_KEY_LENGTH = 50
```

Comment:
```suggestion
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
[security.password_hash]
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;
;; Every parameter for the built in hash algorithms can be changed in this section.
;; Handle with care! Changing the values can massively change the hash calculation time and/or memory needed for the hashing process.
;; After changing a value, the user password hash will be recalculated on next successful login.
;;
;; Parameters for BCrypt
;BCRYPT_COST = 10
;;
;; Parameters for scrypt
;SCRYPT_N = 65536
;SCRYPT_R = 16
;SCRYPT_P = 2
;SCRYPT_KEY_LENGTH = 50
;;
;; Parameters for Argon2id
;ARGON2_ITERATIONS = 2
;ARGON2_MEMORY = 65536
;ARGON2_PARALLELISM = 8
;ARGON2_KEY_LENGTH = 50
;;
;; Parameters for Pbkdf2
;PBKDF2_ITERATIONS = 10000
;PBKDF2_KEY_LENGTH = 50
```

Reply:
But then we need to check the availability of the config and deny usage if certain boundaries are not met. Alternatively we could have a fallback config in place if they are not present -- in fact we _have_ a fallback config at hand for BC.

I would opt in for using the current/old defaults (aka fallbacks). If you're okay I'll try to fix not later than Tuesday...

PS: If we just change as suggest the crypt would be weak a heck if any of the used libs allow `0`s as their params...

Reply:
yes we **should** have a default config and only use app.ini to change it